### ISSUE TITLE
Depending on how you call Pin.Instantiate, Unity may crash.

### ISSUE THREAD

**Issue by tanitaka-tech** [cathei/PinInject#4]
**Describe the bug**

`Pin.Instantiate` a GameObjectContext as a child of a GameObjectContext. 
At this time, if ContextConfiguration is specified as the argument of Instantiate, Unity will crash.

Below is the actual code where the actual crash occurs.
```csharp
public class FieldInstaller : MonoBehaviour, IInjectionContext
{
        [SerializeField] FieldTileFactory _fieldTileFactory;

        void IInjectionContext.Configure(DependencyBinder binder)
        {
            _fieldTileFactory.Inject(createFunc: (prefab, fieldPositionId) =>
            {
                var view = Pin.Instantiate(prefab, transform, fieldGridTileBinder =>
                {
                    fieldGridTileBinder.Bind(fieldPositionId);
                });
                return view;
            });
            binder.Bind<IFieldTileFactory>(_fieldTileFactory);
            ));
        }
}
```

Upon tracing the code for Pin.Instantiate, it appears that a circular reference is created in the Inject method of UnityInjectionStrategy where the parent of the localContainer is set to the localContainer itself.

After entering the following and updating `baseContainer = localContainer;`,
https://github.com/cathei/PinInject/blob/7258e21a4b84e86c579bd94d256fd08ae8ecb966/Runtime/Core/Unity/UnityInjectionStrategy.cs#L24-L36

In the following, `localContainer.SetParent(node.parent != null ? node.parent.Container : baseContainer);` sets the baseContainer (which is localContainer) as the parent of the localContainer.
https://github.com/cathei/PinInject/blob/7258e21a4b84e86c579bd94d256fd08ae8ecb966/Runtime/Core/Unity/UnityInjectionStrategy.cs#L43-L52

**Environment:**

- OS: macOS Ventura13.4.1
- Unity: 2021.3.16f1
- PinInject: 0.1.4

**Expected behaviour**

- Unity does not crash.
- The parameters specified in the ContextConfiguration of Pin.Instantiate are injected into the created GameObjectContext.

**Additional Feedback**

I've used DI container libraries like Zenject, VContainer, and Reflex in the past, but PinInject, which allows me to explicitly specify Dependencies for each context, was perfect for my project. Thank you so much for creating it.

**Comment by cathei** [comment]
Thank you very much for finding and debugging this issue! This should be fixed in [v0.1.5](https://github.com/cathei/PinInject/tree/v0.1.5).

I'm glad that you found this project useful, I appreciate for trying it out! Please reach out if you encounter other problems.

**Comment by tanitaka-tech** [comment]
I was able to confirm the fix on my end as well!
Thank you for your prompt response.
